File: Proj/texasHoldem_Ver1.5/Player.js
/*
 * File:   Player.js
 * Author: Byoung Mo Lee
 * Created on October 11, 2019 02:24 AM
 * Purpose: Texas Holdem
 * 
 */

function Player(){
    this.setPName(name);
    this.nBalance=1000;
    this.myCards=new Card;
    this.active=1;
    this.bet();
    this.check();
    this.fold();
    
}


Player.prototype.setPName=function(name){
    if (name === undefined) {
        this.pName = "";
    }else{this.pName=name;}
};

Player.prototype.bet=function(b1){
    if(b1===undefined) {b1=0;}
    else{this.nBalance-=b1;}
};

Player.prototype.check=function(a1){
    if(a1===undefined) {a1=0;}
    else{this.nBalance-=a1;}this.nBalance-=a1;
};

Player.prototype.fold=function(){
    this.active=0;
};

File: Proj/proj1_texasHoldem/C++/Proj2/texasHoldem_Ver5/Player.cpp
/* 
 * File:   Player.cpp
 * Author: Byoung Mo Lee
 * Created on May 27, 2019 10:12 AM
 * Purpose: Player class for Texas Holdem
 * 
 */

#include "Player.h"

int Player::getNumCards() const{
    int scan=0;
    bool found=0;
    
    while(!found && scan<7){
        if(myCards[scan].getFace()==-1) found=1;
        else scan++;
    }
    
    return scan;
}

void Player::addCards(Card *c, int n){ //mC=players.myCards, c=decked Cards, n=number of addition
    int pos=0;
    bool found=0;
    while(!found){
        if(myCards[pos].getFace()==-1){

            found=1;
        }
        else{
            pos++;
        }
    }
    //cout << myCards[pos].getFace() << endl; // for test
    //cout << pos << endl;                    // for test
    for(int i=0;i<n;i++){
        myCards[pos+i].setACard(c[i]);
        //c->displayCard();  //test for addCard function
    }
}

void Player::print() const{
    
}

